distributed/worker.py

```diff
@@ -3526,11 +3529,28 @@ def _update_who_has(
                 self.has_what[worker].add(key)
                 if ts.state == "fetch":
                     self.data_needed_per_worker[worker].push(ts)
+                    # All workers which previously held a replica of the key are either
+                    # in flight or busy. Kick off ensure_communicating to try fetching
+                    # the data from the new worker.
```

Comment:
```suggestion
                    # We were holding back from fetching this key, possibly because we
                    # didn't know of any workers holding it (that weren't busy, or didn't
                    # already have in-flight data requests).
                    # In case having a new worker available changes things,
                    # kick off ensure_communicating to try fetching the data
                    # from this new worker.
```
There are a few other reasons why the task might still be in `fetch`
1. we're over the `total_out_connections` limit
2. we're over the `comm_threshold_bytes` limits
1. we're paused

Just to make it clear that we're being a little optimistic here, not necessarily testing for all cases in which `ensure_communicating = True` will actually have an effect (which is fine).

Reply:
Updated comment